File: app/core/schemas/task.py
import uuid
from datetime import datetime, date
from pydantic import BaseModel, ConfigDict
from app.enums.task import TaskStatus, TaskPriority
from app.core.schemas.user import UserRead


class TaskRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    title: str
    description: str | None
    status: TaskStatus
    priority: TaskPriority
    position: int
    due_date: date | None
    created_at: datetime
    updated_at: datetime
    creator: UserRead | None
    assignee: UserRead | None


class TaskHistoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    field_changed: str
    old_value: str | None
    new_value: str | None
    created_at: datetime
    user: UserRead | None


class TaskCreate(BaseModel):
    title: str
    description: str | None = None
    priority: TaskPriority = TaskPriority.medium
    assignee_id: uuid.UUID | None = None
    due_date: date | None = None


class TaskUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    status: TaskStatus | None = None
    priority: TaskPriority | None = None
    assignee_id: uuid.UUID | None = None
    due_date: date | None = None


class TaskPositionUpdate(BaseModel):
    position: int

File: frontend-site/src/types/task.ts
export type TaskStatus = 'backlog' | 'todo' | 'in_progress' | 'review' | 'done'
export type TaskPriority = 'low' | 'medium' | 'high' | 'critical'

export interface TaskUser {
  id: string
  email: string
  display_name: string
}

export interface Task {
  id: string
  project_id: string
  title: string
  description: string | null
  status: TaskStatus
  priority: TaskPriority
  position: number
  due_date: string | null
  created_at: string
  updated_at: string
  creator: TaskUser | null
  assignee: TaskUser | null
}

export interface TaskHistoryEntry {
  id: string
  field_changed: string
  old_value: string | null
  new_value: string | null
  created_at: string
  user: TaskUser | null
}

export const STATUS_LABELS: Record<TaskStatus, string> = {
  backlog:     'Бэклог',
  todo:        'К выполнению',
  in_progress: 'В работе',
  review:      'На проверке',
  done:        'Готово',
}

export const PRIORITY_LABELS: Record<TaskPriority, string> = {
  low:      'Низкий',
  medium:   'Средний',
  high:     'Высокий',
  critical: 'Критический',
}

export const PRIORITY_COLORS: Record<TaskPriority, string> = {
  low:      'bg-gray-400',
  medium:   'bg-yellow-400',
  high:     'bg-orange-500',
  critical: 'bg-red-500',
}

export const COLUMNS: TaskStatus[] = ['backlog', 'todo', 'in_progress', 'review', 'done']